the-collab-lab/tcl-75-smart-shopping-list
# Gb db mark as purchased

## Description

This code renders a checkbox for each item in the list. The user can click the checkbox to mark the item as purchased in Firebase, as well as increment the total of that specific item that has been purchased. Items are unchecked 24 hours after they are purchased, to allow the user to purchase them again. 

## Related Issue

closes #9 

## Acceptance Criteria


## Type of Changes

enhancement

## Updates

### Before

<img width="1144" alt="Screenshot 2024-09-05 at 7 48 24 AM" src="https://github.com/user-attachments/assets/3519854a-5e41-44fe-964b-42ecb3499c91">


### After

Below is photo of a list with "apples" checked off
<img width="1440" alt="Screenshot 2024-09-05 at 1 30 59 PM" src="https://github.com/user-attachments/assets/2762a2f5-61f9-45d5-95cd-c3b9018c7d60">

Here you can see that when "apples" was checked as purchased, it set the next purchase date to 24 hours from the purchase date.
<img width="495" alt="Screenshot 2024-09-05 at 1 31 11 PM" src="https://github.com/user-attachments/assets/7a259af0-9ac1-4052-86bb-0e3347568ae5">

## Testing Steps / QA Criteria

1. click checkbox next to item
2. check firebase to make sure item was updated with timestamp, new total purchased quantity and next purchase date (should be 24 hrs from last purchase date)
3. refresh the page to ensure the checkbox stays checked
